### PR TITLE
Feat/Delete Users

### PR DESCRIPTION
**Pull request by gregorysa**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/50)

- Updated DELETE manager to hard-delete the associated Supabase auth user
- Updated DELETE guardian to hard-delete the associated Supabase auth user
- When a guardian is deleted, all their children are deleted (cascading delete)
- When a guardian is deleted, the guardian and child foreign keys in all associated registrations are set to null, and the registration status is set to 'cancelled' if the event has not yet occurred
- Created a transaction in each delete handler so that the database user and Supabase auth user are always deleted together

**Other fixes/additions**:
- Updated auth_id field in User to be required
- Updated auth_id field to be UUID instead of text
- Removed POST endpoints for Guardian and Manager because only the Register endpoints should create new users
- Removed CreateGuardian and CreateManager handlers (only the repository methods are needed)
- Removed all tests associated with the Create Guardian/Manager handlers and routes
- When a guardian is deleted, the guardian foreign keys in all associated reviews are set to null; the review is not deleted
- Updated guardian_id field in Review to be nullable

# How Has This Been Tested?
Edge cases tested:
- Delete transactions will roll back if the Supabase auth user cannot be deleted
- Once a guardian/manager is deleted, the associated ID is no longer in the database
- Once a guardian/manager is deleted, the same user info can be used to sign up again

## Screenshots
Manager is successfully deleted:
Screenshot 2026-02-09 at 1 27 40 AM
Screenshot 2026-02-09 at 1 27 54 AM

Guardian is successfully deleted:
Screenshot 2026-02-09 at 1 22 13 AM
Screenshot 2026-02-09 at 1 22 22 AM

# Checklist

## General
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: backend/main
Comment (adescoteaux1) [inline backend/main]: I dont think this should be here

**Comment by adescoteaux1** [comment]
i requested co-pilot bc there's some things that i was interested in seeing what it thought, but im going through all the comments and will mark resolved ones that we don't need to worry about

File: backend/internal/supabase/migrations/20260203034949_alter-child-registration-tables.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
+ALTER TABLE registration
+ALTER child_id DROP NOT NULL;
+
+ALTER TABLE registration
+ALTER guardian_id DROP NOT NULL;
```
[the remark this answers is not in the document]
Comment (adescoteaux1) [reply backend/internal/supabase/migrations/20260203034949_alter-child-registration-tables.sql]: If this works locally then feel free to ignore!

File: backend/internal/storage/postgres/schema/guardian/sql/get_by_id.sql [hunk 2]
```diff
@@ -1,4 +1,4 @@
-SELECT g.id, g.user_id, u.name, u.email, u.username, u.profile_picture_s3_key, u.language_preference, g.created_at, g.updated_at
+SELECT g.id, g.user_id, u.name, u.email, u.username, u.profile_picture_s3_key, u.language_preference, u.auth_id, g.created_at, g.updated_at
 FROM guardian g
 JOIN "user" u ON g.user_id = u.id
 WHERE g.id = $1
```
[the remark this answers is not in the document]
Comment (adescoteaux1) [reply backend/internal/storage/postgres/schema/guardian/sql/get_by_id.sql]: So sorry I missed in the previous PR. would you be able to update the field type to UUID in the db. It can just be added to one of your current migrations

**Comment by adescoteaux1** [comment]
Just one thing, otherwise looks good!
https://github.com/GenerateNU/skillspark/pull/72#discussion_r2804688006
